akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -63,6 +72,58 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
   def wrapTrailingHeaders(headers: ParsedHeadersFrame): Option[ByteString] = None
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[http2] object ConfigurablePing {
+  case object Tick
+  val Ping = PingFrame(false, ByteString("abcdefgh"))
+  trait PingState {
+    def onDataFrameSeen(): Unit
+    def onPingAck(): Unit
+    def onTick(): Unit
+    def clear(): Unit
+    def shouldEmitPing(): Boolean
+    def sendingPing(timestamp: Long): Unit
+    def pingAckOverdue(timestampNanos: Long): Boolean
+  }
+  object DisabledPingState extends PingState {
+    def onDataFrameSeen(): Unit = ()
+    def onPingAck(): Unit = ()
+    def onTick(): Unit = ()
+    def clear(): Unit = ()
+    def shouldEmitPing(): Boolean = false
+    def sendingPing(timestamp: Long): Unit = ()
+    def pingAckOverdue(timestampNanos: Long): Boolean = false
+  }
+  final class EnabledPingState(pingEveryNTickWithoutData: Long, maxKeepaliveTimeoutNanos: Long) extends PingState {
+    def this(settings: Http2CommonSettings) = this(settings.pingInterval.toSeconds, settings.pingTimeout.toNanos)
```

Comment:
Using `settings.pingInterval.toSeconds` directly works because the scheduler will send a ping per second. If/when we change that `this(settings.pingInterval.toSeconds...` will set the wrong value on `pingEveryNTickWithoutData`.

Maybe adding a constant on  the `case object Tick` to hold that `1.second` and a comment re this relation between schedule period, `pingEveryNTickWithoutData` and `settings.pingInterval` will be enough.